test/ctia/test_helpers/core_test.clj

```diff
@@ -0,0 +1,67 @@
+(ns ctia.test-helpers.core-test
+  (:require [clojure.test :refer [deftest is testing]]
+            [ctia.test-helpers.core :as sut])
+  (:import [clojure.lang ExceptionInfo]
+           [java.util UUID]))
+
+(deftest build-transformed-init-config-test
+  (assert (empty? @#'sut/*config-transformers*))
+  (assert (empty? @#'sut/*properties-overrides*))
+  (testing "defaults to ctia-default.properties values"
+    (is (= (get-in (sut/build-transformed-init-config)
+                   [:ctia :auth :type])
+           :allow-all))
+    (is (= (get-in (sut/build-transformed-init-config)
+                   [:ctia :encryption :type])
+           :default)))
+  (testing "overrides using with-properties"
+    (sut/with-properties
+      ["ctia.auth.type" "foobar"]
+      (is (= (get-in (sut/build-transformed-init-config)
+                     [:ctia :auth :type])
+             :foobar))))
+  (testing "overrides using with-config-transformer*"
+    (sut/with-config-transformer*
+      #(assoc-in % [:ctia :auth :type] :foobar)
+      #(is (= (get-in (sut/build-transformed-init-config)
+                      [:ctia :auth :type])
+              :foobar))))
+  (testing "overrides using with-properties and with-config-transformer*"
+    (testing "both applied with disjoint paths"
```

Comment:
I have trouble understanding what `both applied with disjoint paths` means, probably it can be rewritten to feel more like a complete sentence